src/v/kafka/server/handlers/handlers.h

```diff
@@ -107,4 +110,95 @@ static constexpr size_t max_api_key(type_list<RequestTypes...>) {
+  event_type_entry<
+    create_partitions_api,
+    security::audit::event_type::management>,
+  event_type_entry<create_topics_api, security::audit::event_type::management>,
+  event_type_entry<delete_acls_api, security::audit::event_type::management>,
+  event_type_entry<delete_groups_api, security::audit::event_type::management>,
+  event_type_entry<delete_records_api, security::audit::event_type::management>,
+  event_type_entry<delete_topics_api, security::audit::event_type::management>,
+  event_type_entry<
+    incremental_alter_configs_api,
+    security::audit::event_type::management>,
+  event_type_entry<offset_delete_api, security::audit::event_type::management>,
+  event_type_entry<
+    add_partitions_to_txn_api,
+    security::audit::event_type::produce>,
+  event_type_entry<end_txn_api, security::audit::event_type::produce>,
+  event_type_entry<init_producer_id_api, security::audit::event_type::produce>,
+  event_type_entry<produce_api, security::audit::event_type::produce>,
+  event_type_entry<
+    add_offsets_to_txn_api,
+    security::audit::event_type::consume>,
+  event_type_entry<fetch_api, security::audit::event_type::consume>,
+  event_type_entry<join_group_api, security::audit::event_type::consume>,
+  event_type_entry<leave_group_api, security::audit::event_type::consume>,
+  event_type_entry<list_offsets_api, security::audit::event_type::consume>,
+  event_type_entry<offset_commit_api, security::audit::event_type::consume>,
+  event_type_entry<offset_fetch_api, security::audit::event_type::consume>,
+  event_type_entry<sync_group_api, security::audit::event_type::consume>,
+  event_type_entry<txn_offset_commit_api, security::audit::event_type::consume>,
+  event_type_entry<describe_acls_api, security::audit::event_type::describe>,
+  event_type_entry<describe_configs_api, security::audit::event_type::describe>,
+  event_type_entry<describe_groups_api, security::audit::event_type::describe>,
+  event_type_entry<
+    describe_log_dirs_api,
+    security::audit::event_type::describe>,
+  event_type_entry<find_coordinator_api, security::audit::event_type::describe>,
+  event_type_entry<list_groups_api, security::audit::event_type::describe>,
+  event_type_entry<
+    list_partition_reassignments_api,
+    security::audit::event_type::describe>,
+  event_type_entry<metadata_api, security::audit::event_type::describe>,
+  event_type_entry<
+    offset_for_leader_epoch_api,
+    security::audit::event_type::describe>,
+  event_type_entry<
+    describe_producers_api,
+    security::audit::event_type::describe>,
+  event_type_entry<
+    describe_transactions_api,
+    security::audit::event_type::describe>,
+  event_type_entry<
+    list_transactions_api,
+    security::audit::event_type::describe>,
+  event_type_entry<heartbeat_api, security::audit::event_type::heartbeat>>;
+
+template<typename T>
+constexpr auto get_audit_event_type
+  = event_type_mapper::get<typename std::remove_cvref_t<T>::api_type>::val;
```

Comment:
this is cool, but the thing to be careful about is that we don't make the mistake of generating too much code and actually causing things to run slower. what i mean is that this doesn't seem like a compile time generated map so much as any code that uses it will be duplicated, once for each combination of types being looked up.

Reply:
i suspect it is ok in the cases its used, just something to consider in general.